File: T3/T3_Dialogos/app/src/main/java/com/system/develop/t3_dialogos/dialogos/DialogoListaMultiple.java
package com.system.develop.t3_dialogos.dialogos;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.util.Log;

public class DialogoListaMultiple extends DialogFragment {

    OnDialogoListaListener listener;
    Context context;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;
        try{
            listener = (OnDialogoListaListener) context;
        }catch (ClassCastException e){
            Log.v("error","error en el casteo");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        final String[] opciones = {"sdfsdf","sdfsdfsdf","qrwtghgnfbv"};

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("TITULO LISTA MULTIPLE");
        //builder.setMessage("asdasd");
        builder.setMultiChoiceItems(opciones, null, new DialogInterface.OnMultiChoiceClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which, boolean isChecked) {

            }
        });
        builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {

            }
        });

        //builder.setMu
        return builder.create();
    }

    public interface OnDialogoListaListener{
        public void onElementoListaSelected(String s);
    }
}

File: Kotlin/T5/codigos_ces/Dialogos/app/src/main/java/com/example/dialogos/ui/dialog/DialogoListaMultiple.kt
package com.example.dialogos.ui.dialog

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.example.dialogos.MainActivity

class DialogoListaMultiple : DialogFragment() {

    private lateinit var listener: OnDialogoMultipleListener
    private var posicion: Int? = null
    private var opcion: String? = null


    override fun onAttach(context: Context) {
        super.onAttach(context)
        // listener = context as MainActivity
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        var listaOpciones = arrayOf("Opcion 1", "Opcion 2", "Opcion 3", "Opcion 4")
        var builder: AlertDialog.Builder = AlertDialog.Builder(requireActivity())

        builder.setTitle("Selecciona la opcion correcta")

        builder.setMultiChoiceItems(listaOpciones, null) { _, i, b ->
            // se agrega el elemento en el arraylist, solo cuando pase a b true
            // se elimina el elemento en el arraylist, solo cuando pasa a b false
        }

        builder.setNeutralButton("Cancelar") { _, _ ->
            dismiss()
        }
        builder.setPositiveButton("Aceptar") { _, _ ->
            //dismiss()
            // comunico el arraylist completo
        }

        return builder.create()
    }

    interface OnDialogoMultipleListener {
        fun onOpcionMultpleSelected(respuestas: ArrayList<CharSequence>)
    }


}